### PR TITLE
Overhaul color: fix bugs, change to 0-1

### PR DESCRIPTION
**Pull request by idbrii**
Bug fixed: 
* 6eebed814c012c023cbba8cb72b490b031501c7b -- `c + c`, `c * c`, `c:is_color()`, ... gave errors -- #34.
* c5fe950045a2af53d1f732d856cc03a21e045389 -- `color():lighten(10)` would be white because the logic didn't match the docs that say the input is 0-255.
* f7903ad0d264a3b702660fb0c2ebf2af6213db0a -- hsv conversion functions were producing numbers in ranges that didn't match the color inputs.

The 0,1 change is a pretty big one, but there were some major issues already so this should be a big improvement. Legacy 255 range users will have to input values normalized but can get output values in 0-255.

```lua
local c = color(255/255, 120/255, 100/255, 255/255)
c:hue(100)
something:setColor(c:as_255())
```

### CODE REVIEW

File: modules/color.lua [hunk 1]
```diff
@@ -305,14 +328,12 @@ function color.linear_to_gamma(r, g, b, a)
 
 	if type(r) == "table" then
 		local c = {}
-		for i = 1, 3 do
-			c[i] = convert(r[i] / 255) * 255
+		for i = 1, 4 do
+			c[i] = convert(r[i])
```
Comment (shakesoda) [inline modules/color.lua:332]: this is suspect (was before, too!), alpha is intended to always be linear in all paths. probably my screwup, c[4] = r[4] was the intent.

Comment (idbrii) [reply modules/color.lua:334]: Ah, the original code makes more sense now. Fixed!

**Comment by idbrii** [comment]
Since this PR is changing from [0,255] to [0,1], should hue change from [0,360] to [0,1]?

imgui shows 255 even for hue in the UI, but [uses 0-1 for everything internally](https://github.com/ocornut/imgui/blob/master/imgui.cpp#L2045).

Unity [uses H, S, and V in the range 0.0 to 1.0](https://docs.unity3d.com/ScriptReference/Color.RGBToHSV.html). So [does Godot](https://docs.godotengine.org/en/stable/classes/class_color.html#class-color-property-h).

So it seems like there's some amount of precedence for that.

**Comment by idbrii** [comment]
Pushed new changes to convert hue to 0,1 and more extensive hsv tests.

I also noticed a bug where passing equal r,g,b values (i.e., white) to color_to_hsv would produce a nan value for hue. This is also [an issue with the reference C++ code](
https://tio.run/##Sy4o0E3OScxL//9fOTMvOac0JVXBJjO/uKQoNTHXjiszr0QhNzEzT0NToZqLs7gkxcoqOb@0RMHGRkHJIzUnJ19HITy/KCdFUQkkBJZPzUvJseaq/f8fAA). I added a new commit to fix that.

**Pull request merged**